vaadin-testbench-unit/src/main/kotlin/com/vaadin/testbench/unit/mocks/MockedUI.kt

```diff
@@ -9,10 +9,27 @@
  */
 package com.vaadin.testbench.unit.mocks
 
+import com.vaadin.flow.component.Component
+import com.vaadin.flow.component.ComponentEventListener
 import com.vaadin.flow.component.UI
+import com.vaadin.flow.shared.Registration
+import java.util.concurrent.atomic.AtomicReference
 
 /**
  * A simple no-op UI used by default by [com.vaadin.testbench.unit.MockVaadin.setup].
  * The class is open, in order to be extensible in user's library
  */
-open class MockedUI : UI()
\ No newline at end of file
+open class MockedUI : UI() {
+
+    override fun setChildComponentModal(childComponent: Component?, modal: Boolean) {
+        super.setChildComponentModal(childComponent, modal)
+        if (modal) {
```

Comment:
Is a round trip not needed when child is set to be modeless?

Reply:
makes sense to execute also for that. Added roundtrip and removed it from ComponentWrap::setModal